app/queries/integration_items_query.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class IntegrationItemsQuery < BaseQuery
+  def call(search_term:, integration_id:, page:, limit:, filters: {})
+    @search_term = search_term
+    @integration_id = integration_id
+
+    integration_items = base_scope.result
+    integration_items = integration_items.where(integration_id:) if integration_id.present?
+    integration_items = integration_items.where(item_type: filters[:item_type]) unless filters[:item_type].nil?
+    integration_items = integration_items.order(name: :asc).page(page).per(limit)
+
+    result.integration_items = integration_items
+    result
+  end
+
+  private
+
+  attr_reader :search_term
+
+  def base_scope
+    IntegrationItem.ransack(search_params)
```

Comment:
We have to include organization in the query as well. Also, I would prefer having `integration_id` here as well, so that in the main method we can only have filters + pagination

Reply:
I added the organization.